optaplanner-examples/src/main/java/org/optaplanner/examples/meetingscheduling/persistence/MeetingSchedulingXlsxFileIO.java

```diff
@@ -0,0 +1,825 @@
+                List<MeetingAssignment> meetingAssignmentList = roomMeetingAssignmentList.stream()
+                        .filter(meetingAssignment -> meetingAssignment.getStartingTimeGrain() == timeGrain)
+                        .collect(toList());
+                if (meetingAssignmentList.isEmpty() && mergingPreviousTimeGrain && previousMeetingRemainingTimeGrains > 0) {
+                    previousMeetingRemainingTimeGrains--;
+                    nextCellVertically();
+                } else {
+                    if (mergingPreviousTimeGrain && mergeStart < currentRowNumber) {
+                        currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, currentColumnNumber, currentColumnNumber));
+                    }
+
+                    StringBuilder meetingInfo = new StringBuilder();
+                    for (MeetingAssignment meetingAssignment : meetingAssignmentList) {
+                        String startTimeString = getTimeString(meetingAssignment.getStartingTimeGrain().getStartingMinuteOfDay());
+                        String endTimeString = getTimeString(solution.getTimeGrainList().get(meetingAssignment.getLastTimeGrainIndex()).getStartingMinuteOfDay()
+                                                                     + TimeGrain.GRAIN_LENGTH_IN_MINUTES);
+                        meetingInfo.append(StringUtils.abbreviate(meetingAssignment.getMeeting().getTopic(), 150)).append("\n  ")
+                                .append(meetingAssignment.getMeeting().getSpeakerList().stream().map(Person::getFullName).collect(joining(", "))).append("\n  ")
+                                .append(startTimeString).append(" - ").append(endTimeString)
+                                .append(" (").append(meetingAssignment.getMeeting().getDurationInGrains() * TimeGrain.GRAIN_LENGTH_IN_MINUTES).append(" mins)");
+                    }
+                    nextCellVertically().setCellValue(meetingInfo.toString());
+
+                    previousMeetingRemainingTimeGrains = getLongestDurationInGrains(meetingAssignmentList) - 1;
+                    mergingPreviousTimeGrain = previousMeetingRemainingTimeGrains > 0;
+                    mergeStart = currentRowNumber;
+                }
+            }
+            if (mergeStart < currentRowNumber) {
+                currentSheet.addMergedRegion(new CellRangeAddress(mergeStart, currentRowNumber, currentColumnNumber, currentColumnNumber));
+            }
+        }
+
+        private String getTimeString(int minuteOfDay) {
+            return TIME_FORMATTER.format(LocalTime.ofSecondOfDay(minuteOfDay * 60));
+        }
+
+        private void writeMeetingAssignmentList(List<MeetingAssignment> meetingAssignmentList) {
+            String[] filteredConstraintNames = Arrays.stream(MeetingSchedulingConstraints.values())
+                    .map(MeetingSchedulingConstraints::getConstraintName)
+                    .toArray(String[]::new);
```

Comment:
I think it would be easier to change the signature of `nextMeetingAssignmentListCell()` to accept a Collection. Currently, the values set is converted to an array and then, inside the `nextMeetingAssignmentListCell()` method, converted from an array to list. With the Collection interface - no conversion at all :)